smart_contract/executor/src/default.rs

```diff
@@ -1168,6 +1465,24 @@ pub mod trigger {
 
         deny!(executor, "Can't execute trigger owned by another account");
     }
+
+    fn is_token_trigger_associated(permission: &PermissionToken, trigger_id: &TriggerId) -> bool {
+        tokens::trigger::CanUnregisterUserTrigger::try_from(permission.clone())
+            .map(|permission| &permission.trigger_id == trigger_id)
+            .or_else(|_err| {
+                tokens::trigger::CanBurnUserTrigger::try_from(permission.clone())
+                    .map(|permission| &permission.trigger_id == trigger_id)
+            })
+            .or_else(|_err| {
+                tokens::trigger::CanExecuteUserTrigger::try_from(permission.clone())
+                    .map(|permission| &permission.trigger_id == trigger_id)
+            })
+            .or_else(|_err| {
+                tokens::trigger::CanMintUserTrigger::try_from(permission.clone())
+                    .map(|permission| &permission.trigger_id == trigger_id)
+            })
+            .unwrap_or(false)
+    }
```

Comment:
can't you use `map_token` macro?

Reply:
i don't think so, `map_token` requires uniform `callback` for every token which is not the case here

Reply:
how is it not the case? it looks uniform to me

Reply:
`map_token!` works with every default token and if parsed successfully call `callback` on it

in my case i only try small subset of tokens

Reply:
got it, then at least use match statement like in the `map_token!` macro